src/swarm-engine/Swarm/Game/Location.hs

```diff
@@ -205,6 +207,17 @@ manhattan (Location x1 y1) (Location x2 y2) = abs (x1 - x2) + abs (y1 - y2)
 euclidean :: Location -> Location -> Double
 euclidean p1 p2 = norm (fromIntegral <$> (p2 .-. p1))
 
+-- | Get all the locations that are within a certain manhattan
+--   distance from a given location.
+--
+-- >>> getLocsInArea (P (V2 0 0)) 1
+-- [P (V2 0 0), P (V2 0 1), P (V2 0 (-1)), P (V2 1 0), P (V2 1 1), P (V2 1 (-1))]
+-- >>> map (\i -> length (getLocsInArea origin i)) [0..8]
+-- [1,5,13,25,41,61,85,113,145]
+getLocsInArea :: Location -> Int32 -> [Location]
+getLocsInArea loc r =
+  [loc .+^ V2 dx dy | x <- [0 .. 5], y <- [0 .. r - x], dx <- nub [x, -x], dy <- nub [y, -y]]
```

Comment:
> `x <- [0 .. 5]`

Where does the number `5` come from?

Reply:
May also be worth making mention of `Swarm.Game.Step.Const.genDiamondSides`.

Reply:
> Where does the number `5` come from?

Oops!  Good catch, I abstracted this from some specific code I wrote for teleport, and missed changing that 5 to `r`.  I wonder why the tests passed?  Maybe those doctests aren't getting run or something?

Thanks for the reference to `genDiamondSides`, I'll add a note.